### ISSUE TITLE
`path` in `listDirectory` result record is `null`

### ISSUE THREAD

**Issue by icidasset** [gren-lang/node#21]
When trying to list a directory with only a directory in it, the `path` property of the resulting record is `null` and crashes the Node program (discovered using `Debug.log`)

Looks like the `name` property is returned in kernel code instead of `path`: https://github.com/gren-lang/node/blob/b2a184b/src/Gren/Kernel/FileSystem.js#L229

Crash error:
```
/Users/steven/Code/icidasset/shikensu-gren/example/generate:1773
				}, A2($gren_lang$core$Array$append, right.directory, A2($gren_lang$core$Array$pushLast, $gren_lang$node$FileSystem$Path$filenameWithExtension(left), left.directory))), extension: right.extension, filename: right.filename });
				                                          ^

TypeError: Cannot read properties of undefined (reading 'directory')
    at Function.f (/Users/steven/Code/icidasset/shikensu-gren/example/generate:1773:47)
```
When trying use the "null" path with `Path.prepend`

**Comment by robinheghan** [comment]
I'm not able to reproduce this.

I tried this:

```
main : Node.SimpleProgram msg
main =
    Node.defineSimpleProgram init


init : Node.Environment -> Init.Task ( Cmd msg )
init env =
    Init.await FileSystem.initialize <| \fs ->
        FileSystem.listDirectory fs (Path.fromPosixString "dir")
            |> Task.map (Debug.toString)
            |> Task.onError (Task.succeed << FileSystem.errorToString)
            |> Task.andThen (Stream.sendLine env.stdout)
            |> Task.execute
            |> Node.endWithCmd
```

where `dir` only contains a `dir1` subdirectory.

Worked as expected.

**Comment by robinheghan** [comment]
Fixed in 4.1.1
